pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -244,3 +249,20 @@ func (c *Reconciler) getServiceAccount(pr *v1alpha1.PipelineRun) (string, error)
 	}
 	return pp.Spec.ServiceAccount, nil
 }
+
+func (c *Reconciler) reconcilePipelineRunStatus(pr *v1alpha1.PipelineRun, state []*resources.PipelineRunTaskRun) {
+	// reconcile pipeline run status first to see if we need to run any more tasks.
+	prStatus := resources.GetPipelineConditionStatus(pr.Name, state, c.Logger)
+	pr.Status.SetCondition(prStatus)
+	c.Logger.Infof("PipelineRun %s status is being set to %s", pr.Name, pr.Status)
```

Comment:
hm i think this will conflict with https://github.com/knative/build-pipeline/pull/176 but whichever one of us merges second can deal with it :)

Reply:
sure. l am probably going to hold this PR until Thursday. I do not want to add the flag like @pivotal-nader-ziada mentioned since this is the default functionality of pipeline and described in the docs.

Reply:
the flag was just a suggestion, but based on the comments in the issue, I'm not sure if you need it